src/zip/zipmacro.rs

```diff
@@ -56,6 +56,11 @@
 ///
 /// type M = Array2<f32>;
 ///
+/// // Since this function borrows its inputs, captures must use the x (x) pattern
+/// fn borrow_multiply(a: &mut M, b: &M, c: &M) {
+///     azip!(mut a (a), b (b), c (c) in { *a = b * c });
+/// }
```

Comment:
If you have time, move this function to below the main function. I think it can be read last.